runtimes/pythonrt/runner/main.py

```diff
@@ -444,21 +444,25 @@ def on_event(self, fn, event):
             runner_id=self.id,
         )
 
-        if result.error:
-            req.error = result_error(result.error)
-            tb = pb_traceback(result.traceback)
-            req.traceback.extend(tb)
-        else:
-            try:
-                data = pickle.dumps(result)
-                req.result.custom.data = data
-                req.result.custom.value.CopyFrom(values.safe_wrap(result.value))
-            except (TypeError, pickle.PickleError) as err:
-                req.error = f"can't pickle {result.value} - {err}"
+        try:
+            if result.error:
+                req.error = result_error(result.error)
+                tb = pb_traceback(result.traceback)
+                req.traceback.extend(tb)
+            else:
+                try:
+                    data = pickle.dumps(result)
+                    req.result.custom.data = data
+                    req.result.custom.value.CopyFrom(values.safe_wrap(result.value))
+                except (TypeError, pickle.PickleError) as err:
+                    req.error = f"can't pickle {result.value} - {err}"
+        except Exception as err:
+            req.error = f"unexpected error: {err}"
+            log.exception("on_event: %r", err)
 
         try:
             self.worker.Done(req)
-        except grpc.RpcError as err:
+        except Exception as err:
```

Comment:
Better to keep specific error types, and distinguish them from unexpected ones:

```suggestion
        except grpc.RpcErr as err:
            log.error("on_event: RPC error when sending done: %r", err)
        except Exception as ex:
            log.error("on_event: unexpected error when sending done: %r", ex)
```

Reply:
there can be lots of other exception types in this case, 
we don't really case in this scenario, 
we're not retrying and it's the end of the runner